File: samples/basic-samples/simple-stack-example-basic-java-view/src/main/java/com/zhuinden/navigationexampleview/screens/HomeView.java
package com.zhuinden.navigationexampleview.screens;

import android.annotation.TargetApi;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.RelativeLayout;

import com.zhuinden.navigationexampleview.R;
import com.zhuinden.simplestack.Backstack;
import com.zhuinden.simplestack.navigator.Navigator;

/**
 * Created by Owner on 2017. 06. 29..
 */

public class HomeView
        extends RelativeLayout /*implements Bundleable*/ {
    public HomeView(Context context) {
        super(context);
    }

    public HomeView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public HomeView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @TargetApi(21)
    public HomeView(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
    }

    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();

        findViewById(R.id.home_button).setOnClickListener(v -> {
            Navigator.getBackstack(getContext()).goTo(OtherKey.create());
        });

        HomeKey homeKey = Backstack.getKey(getContext()); // get args
    }
}

File: samples/basic-samples/simple-stack-example-basic-kotlin-view/src/main/java/com/zhuinden/navigationexamplekotlinview/screens/HomeView.kt
package com.zhuinden.navigationexamplekotlinview.screens

import android.annotation.TargetApi
import android.content.Context
import android.util.AttributeSet
import android.widget.RelativeLayout
import com.zhuinden.navigationexamplekotlinview.databinding.HomeViewBinding
import com.zhuinden.navigationexamplekotlinview.utils.onClick
import com.zhuinden.simplestack.Backstack
import com.zhuinden.simplestackextensions.navigatorktx.backstack

/**
 * Created by Owner on 2017. 06. 29..
 */

class HomeView : RelativeLayout {
    constructor(context: Context) : super(context)
    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs, defStyleAttr)

    @TargetApi(21)
    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes)

    private lateinit var binding: HomeViewBinding

    override fun onFinishInflate() {
        super.onFinishInflate()

        binding = HomeViewBinding.bind(this)

        binding.buttonHome.onClick {
            backstack.goTo(OtherKey())
        }

        val homeKey = Backstack.getKey<HomeKey>(context) // get args
    }
}

